File: DemoServer/AdditionalLanguages/php/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.php
<?php

namespace RavenDB\Demo\textSearch\fTSQuerySearchOperators;

//region Usings
use RavenDB\Documents\Queries\SearchOperator;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Employee;

class FTSQuerySearchOperators
{
    public function __invoke(RunParams $runParams): array
    {
        $term1 = $runParams->getTerm1();
        $term2 = $runParams->getTerm2();
        $term3 = $runParams->getTerm3();

        //region Demo
        $employeesWithMatchingTerms = [];

        $session = DocumentStoreHolder::getStore()->openSession();
        try {
            //region Step_1
            $employeesWithMatchingTerms = $session->query(Employee::class)
            //endregion
                //region Step_2
                ->search("Notes", $term1 . " " . $term2, SearchOperator::and())
                //endregion
                //region Step_3
                ->search("Notes", $term3, SearchOperator::or())
                //endregion
                //region Step_4
                ->toList();
                //endregion
        } finally {
            $session->close();
        }
        //endregion
        return $employeesWithMatchingTerms;
    }
}

File: DemoServer/AdditionalLanguages/python/textSearch/fTSQuerySearchOperators/ftsQuerySearchOperators.py
from demo_example import Example, RunParamsBase
from models import Employee

#region Usings
from ravendb import SearchOperator
#endregion

class RunParams(RunParamsBase):
    def __init__(self, term1: str = None, term2: str = None, term3: str = None):
        self.term1 = term1
        self.term2 = term2
        self.term3 = term3

class FtsQueryOperators(Example):
    def run(self, run_params: RunParams):
        term1 = run_params.term1
        term2 = run_params.term2
        term3 = run_params.term3

        #region Demo
        with self.document_store_holder.store().open_session() as session:
            
            #region Step_1
            my_query = session.query(object_type=Employee)
            #endregion
            #region Step_2
            my_query = my_query.search("Notes", f"{term1} {term2}", SearchOperator.AND)
            #endregion
            #region Step_3
            my_query = my_query.search("Notes", term3)
            #endregion
            
            #region Step_4
            employees_with_matching_terms = list(my_query)
            #endregion
        #endregion
        
        return employees_with_matching_terms
